pkg/executor/test/admintest/admin_test.go

```diff
@@ -1773,3 +1779,219 @@ func TestAdminCheckTableErrorLocateForClusterIndex(t *testing.T) {
+		tk.MustExec("insert admin_test values (-10, -20, 1), (-1, -10, 2), (1, 11, 3), (2, 12, 0), (5, 15, -1), (10, 20, -2), (20, 30, -3)")
+
+		// Make some corrupted index. Build the index information.
+		sctx := mock.NewContext()
+		sctx.Store = store
+		is := domain.InfoSchema()
+		dbName := model.NewCIStr("test")
+		tblName := model.NewCIStr("admin_test")
+		tbl, err := is.TableByName(dbName, tblName)
+		require.NoError(t, err)
+		tblInfo := tbl.Meta()
+		idxInfo := tblInfo.Indices[0]
+		require.True(t, idxInfo.Global)
+		df := tblInfo.GetPartitionInfo().Definitions[0]
+
+		// Reduce one row of table.
+		// Index count > table count, (2, 12, 0) is deleted.
+		txn, err := store.Begin()
+		require.NoError(t, err)
+		txn.Delete(tablecodec.EncodeRowKey(df.ID, kv.IntHandle(4).Encoded()))
+		err = txn.Commit(context.Background())
+		require.NoError(t, err)
+		err = tk.ExecToErr("admin check table admin_test")
+		require.Error(t, err)
+		require.True(t, consistency.ErrAdminCheckInconsistent.Equal(err))
+		require.ErrorContains(t, err, "[admin:8223]data inconsistency in table: admin_test, index: uidx_a, handle: 4, index-values:\"handle: 4, values: [KindInt64 2")
+
+		// Remove corresponding index key/value.
+		// Admin check table will success.
+		idxVal, err := codec.EncodeKey(tk.Session().GetSessionVars().StmtCtx.TimeZone(), nil, types.NewIntDatum(2))
+		require.NoError(t, err)
+		txn, err = store.Begin()
+		require.NoError(t, err)
+		txn.Delete(tablecodec.EncodeIndexSeekKey(tblInfo.ID, idxInfo.ID, idxVal))
+		err = txn.Commit(context.Background())
+		require.NoError(t, err)
+		tk.MustExec("admin check table admin_test")
+
+		// Reduce one row of index.
+		// Index count < table count, (-1, -10, 2) is deleted.
+		idxVal, err = codec.EncodeKey(tk.Session().GetSessionVars().StmtCtx.TimeZone(), nil, types.NewIntDatum(-1))
+		require.NoError(t, err)
+		txn, err = store.Begin()
+		require.NoError(t, err)
+		txn.Delete(tablecodec.EncodeIndexSeekKey(tblInfo.ID, idxInfo.ID, idxVal))
+		err = txn.Commit(context.Background())
+		require.NoError(t, err)
+		err = tk.ExecToErr("admin check table admin_test")
+		require.Error(t, err)
+		require.True(t, consistency.ErrAdminCheckInconsistent.Equal(err))
+		require.EqualError(t, err, "[admin:8223]data inconsistency in table: admin_test, index: uidx_a, handle: 2, index-values:\"\" != record-values:\"handle: 2, values: [KindInt64 -1]\"")
+
+		// Add one row with inconsistent value.
+		// Index count = table count, but data is different.
+		rd := rowcodec.Encoder{Enable: true}
+		value, err := tablecodec.EncodeRow(tk.Session().GetSessionVars().StmtCtx.TimeZone(),
+			[]types.Datum{types.NewIntDatum(100), types.NewIntDatum(200), types.NewIntDatum(300)},
+			[]int64{tblInfo.Cols()[0].ID, tblInfo.Cols()[1].ID, tblInfo.Cols()[2].ID}, nil, nil, &rd)
+		require.NoError(t, err)
+		txn, err = store.Begin()
+		require.NoError(t, err)
+		err = txn.Set(tablecodec.EncodeRowKeyWithHandle(df.ID, kv.IntHandle(2)), value)
```

Comment:
In the above case, we delete the value of idx.  This case sets the row value of IntHandle(2), but should idx cnt still be less than tbl count?

Reply:
Done